### PR TITLE
default scheme

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Contributes to #26682
[Internal review URL](https://review.learn.microsoft.com/en-us/aspnet/core/security/authentication/?view=aspnetcore-7.0&branch=pr-en-us-27310#authentication-concepts)

### CODE REVIEW

**Comment by blowdart** [comment]
@haok because I get this wrong a lot :)

File: aspnetcore/security/authentication/index.md [hunk 1]
```diff
@@ -48,13 +48,22 @@ The Authentication middleware is added in `Program.cs` by calling <xref:Microsof
...
+When there is only a single authentication scheme registered, it becomes the default scheme. If multiple schemes are registered and the default scheme isn't specified, a scheme must be specified in the authorize attribute, otherwise, the following error is thrown:
 
 > InvalidOperationException: No authenticationScheme was specified, and there was no DefaultAuthenticateScheme found. The default schemes can be set using either AddAuthentication(string defaultScheme) or AddAuthentication(Action\<AuthenticationOptions> configureOptions).
 
+### `DefaultScheme`
+
+When there is only a single authentication scheme registered, the single authentication scheme:
+
+* Is automatically is used as the `DefaultScheme`.
+* Eliminates the need to specify the `DefaultScheme` in `AddAuthentication`.
```
Comment (HaoK) [inline aspnetcore/security/authentication/index.md:63]: maybe also mention via `AuthenticationOptions` to?

**Review by HaoK — APPROVED** [review]

File: aspnetcore/security/authentication/index.md (hunk 1 above)
Comment (Rick-Anderson) [reply aspnetcore/security/authentication/index.md]: > maybe also mention via `AuthenticationOptions` too?

@HaoK  can you check [this commit](https://github.com/dotnet/AspNetCore.Docs/pull/27310/commits/b4196a448fa1776480ede3c27579fe67c02fd3d5)

File: aspnetcore/security/authentication/index.md [hunk 2]
```diff
@@ -48,13 +48,22 @@ The Authentication middleware is added in `Program.cs` by calling <xref:Microsof
...
-There's no automatic probing of schemes. If the default scheme isn't specified, the scheme must be specified in the authorize attribute, otherwise, the following error is thrown:
+When there is only a single authentication scheme registered, it becomes the default scheme. If multiple schemes are registered and the default scheme isn't specified, a scheme must be specified in the authorize attribute, otherwise, the following error is thrown:
 
 > InvalidOperationException: No authenticationScheme was specified, and there was no DefaultAuthenticateScheme found. The default schemes can be set using either AddAuthentication(string defaultScheme) or AddAuthentication(Action\<AuthenticationOptions> configureOptions).
 
+### `DefaultScheme`
+
+When there is only a single authentication scheme registered, the single authentication scheme:
+
+* Is automatically is used as the `DefaultScheme`.
```
Comment (serpent5) [inline aspnetcore/security/authentication/index.md:62]:
```suggestion
* Is automatically used as the `DefaultScheme`.
```

File: aspnetcore/security/authentication/index.md [hunk 3]
```diff
@@ -48,13 +48,22 @@ The Authentication middleware is added in `Program.cs` by calling <xref:Microsof
...
+* Eliminates the need to specify the `DefaultScheme` in `AddAuthentication` or [`AddAuthentication(IServiceCollection, Action<AuthenticationOptions>)`](/dotnet/api/microsoft.extensions.dependencyinjection.authenticationservicecollectionextensions.addauthentication#microsoft-extensions-dependencyinjection-authenticationservicecollectionextensions-addauthentication(microsoft-extensions-dependencyinjection-iservicecollection-system-action((microsoft-aspnetcore-authentication-authenticationoptions))))
+
+The single authentication scheme being automatically used as the `DefaultScheme` can be disabled via `AppContext.SetSwitch("Microsoft.AspNetCore.Authentication.SuppressAutoDefaultScheme")`.
```
Comment (serpent5) [inline aspnetcore/security/authentication/index.md:65]:
```suggestion
To disable automatically using the single authentication scheme as the `DefaultScheme`, call `AppContext.SetSwitch("Microsoft.AspNetCore.Authentication.SuppressAutoDefaultScheme")`.
```

File: aspnetcore/security/authentication/index.md [hunk 4]
```diff
@@ -48,13 +48,22 @@ The Authentication middleware is added in `Program.cs` by calling <xref:Microsof
...
 
 > InvalidOperationException: No authenticationScheme was specified, and there was no DefaultAuthenticateScheme found. The default schemes can be set using either AddAuthentication(string defaultScheme) or AddAuthentication(Action\<AuthenticationOptions> configureOptions).
 
+### `DefaultScheme`
+
+When there is only a single authentication scheme registered, the single authentication scheme:
+
+* Is automatically is used as the `DefaultScheme`.
+* Eliminates the need to specify the `DefaultScheme` in `AddAuthentication` or [`AddAuthentication(IServiceCollection, Action<AuthenticationOptions>)`](/dotnet/api/microsoft.extensions.dependencyinjection.authenticationservicecollectionextensions.addauthentication#microsoft-extensions-dependencyinjection-authenticationservicecollectionextensions-addauthentication(microsoft-extensions-dependencyinjection-iservicecollection-system-action((microsoft-aspnetcore-authentication-authenticationoptions))))
```
Comment (serpent5) [inline aspnetcore/security/authentication/index.md:63]:
```suggestion
* Eliminates the need to specify the `DefaultScheme` in `AddAuthentication` or <xref:Microsoft.Extensions.DependencyInjection.AuthenticationCoreServiceCollectionExtensions.AddAuthenticationCore(Microsoft.Extensions.DependencyInjection.IServiceCollection,System.Action{Microsoft.AspNetCore.Authentication.AuthenticationOptions})>.
```

**Comment by HaoK** [comment]
Looks good!

**Comment by Rick-Anderson** [comment]
@serpent5 much appreciated.

**Pull request merged**